zookeeper/core/component.py

```diff
@@ -289,14 +291,20 @@ def _list_field_strings(instance, color: bool, single_line: bool) -> Iterator[st
         except AttributeError as e:
             if field.allow_missing:
                 value = utils.missing
+                yield f"{field_name}={value}" if color else f"{field_name}={value}"
+                continue
             else:
                 raise e from None
 
         parent_instance = next(
             utils.generate_component_ancestors_with_field(instance, field_name), None
         )
         if parent_instance is not None:
-            is_inherited = base_getattr(parent_instance, field_name) == value  # type: ignore
+            parent_value = base_getattr(parent_instance, field_name)  # type: ignore
+            if is_tensor(parent_value):
+                is_inherited = parent_value is value
+            else:
+                is_inherited = parent_value == value
         else:
```

Comment:
@AdamHillier  do we even need this distinction? If it really is inherited, we could always just check `parent_value is value`, right?

Reply:
Hmm I'm slightly unclear on what's going on. What's wrong with always using the `==` check?

Reply:
`tensor == tensor` returns a tensor rather than a boolean, which breaks the logic

Reply:
Ah I see... That's super dumb, but makes sense, thanks :+1: I think I'd be okay with just using `parent_value is value`, actually, as you suggest.